### ISSUE TITLE
Signup: email field in Social First signup is missing accessible name.

### ISSUE THREAD

**Issue by worldomonation** [Automattic/wp-calypso#82624]
### Quick summary

The email field in the social first signup screen is missing an accessible name.

![image](https://github.com/Automattic/wp-calypso/assets/6549265/fd8681ec-e72b-4c94-a719-07ee11b6f79e)

### Steps to reproduce

1. navigate to `/log-in`.
2. click on "create a new account".
3. click on "email".

### What you expected to happen

Every key element on the page has accessible name.

### What actually happened

Email input field is missing an accessible name.

### Impact

All

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

Simple, Atomic
